pkg/kubecost/asset.go

```diff
@@ -2566,12 +2567,31 @@ func (as *AssetSet) Each(f func(string, Asset)) {
 		f(k, a)
 	}
 }
+*/
 
 // End returns the end time of the AssetSet's window
 func (as *AssetSet) End() time.Time {
 	return *as.Window.End()
 }
 
+// Each invokes the given function for each Asset in the set
+func (as *AssetSet) Each(f func(string, Asset)) {
+	if as == nil {
+		return
+	}
+
+	as.RLock()
+	assets := make(map[string]Asset, len(as.assets))
+	for k, v := range as.assets {
+		assets[k] = v
+	}
+	as.RUnlock()
```

Comment:
While I'm not crazy about duplicating the collection here, it at least provides collection safety here. Since deep copy is out of the question due to performance, this is the next best thing.